data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkRetryStrategy.java

```diff
@@ -238,6 +238,13 @@ private BulkOperationRequestResponse handleRetriesAndFailures(final Accumulating
         if (doRetry) {
             if (retryCount % 5 == 0) {
                 LOG.warn("Bulk Operation Failed. Number of retries {}. Retrying... ", retryCount, e);
+                if (Objects.isNull(e)) {
+                    for (final BulkResponseItem bulkItemResponse : bulkResponse.items()) {
+                        if (Objects.nonNull(bulkItemResponse.error())) {
```

Comment:
Do you mean to say "it will be NULL for successful records"?
Don't know how to test this path easily. But based on the following existing code, I think it will be set to null for successful records.

```
if (bulkItemResponse.error() == null) {
                    sentDocumentsOnFirstAttemptCounter.increment();
 }
```